### PR TITLE
fix: detect lambdatest service

### PR DESCRIPTION
**Pull request by Joozty**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
Some comments:

- can we add a test to `packages/webdriverio/tests/utils/detectBackend.test.ts`
- do all LambdaTest keys have a length of 50?
- Can we find a place in `/website/docs` where we document for which vendors WebdriverIO can derive the connection details?

**Comment by Joozty** [comment]
> can we add a test to packages/webdriverio/tests/utils/detectBackend.test.ts

done

> do all LambdaTest keys have a length of 50?

Not sure about that. Mine has 50 characters though. 🤔 


> Can we find a place in /website/docs where we document for which vendors WebdriverIO can derive the connection details?

Not sure where. 🤷 

 Btw. I assumed all services had this feature but I was wrong. Lambdatest [docs page](https://webdriver.io/docs/wdio-lambdatest-service/) doesn't mention anything about specifying the host and other params so maybe this PR is not a feature but a bug fix.

**Comment by christian-bromann** [comment]
> Not sure where. 🤷

It seems that documentation was already added in https://github.com/webdriverio/webdriverio/pull/5097 but the key detection was missing. Not sure if maybe the service added connection details. Anyway, adding it to the core package makes total sense to me.

> so maybe this PR is not a feature but a bug fix.

Makes sense to me.

**Review by christian-bromann — COMMENTED** [review]
The PR looks good, mind moving it out of draft`?

File: packages/webdriverio/src/utils/detectBackend.ts [hunk 1]
```diff
@@ -97,6 +97,19 @@ export default function detectBackend(options: BackendConfigurations = {}) {
         }
     }
 
+    /**
+     * Lambdatest
+     * e.g. cYAjKrqGwyPjPQv41ICDF4C5OjlxzA9epZsnugVJJxqOReWRWU
+     */
+    if (typeof user === 'string' && typeof key === 'string' && key.length === 50) {
```
Comment (mgrybyk) [inline packages/webdriverio/src/utils/detectBackend.ts:104]: Not a concert regarding the PR but it doesn't seem to be a good idea to rely on key length.
Maybe it makes sense to explicitly set a service type? 
cc: @christian-bromann

**Review by mgrybyk — APPROVED** [review]

File: packages/webdriverio/src/utils/detectBackend.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/webdriverio/src/utils/detectBackend.ts:104]: @mgrybyk it is currently the only way to differentiate between vendors. We do the same for others too. At the end of the day you can always provide a target host to connect to.

**Comment by christian-bromann** [comment]
Thanks a lot @Joozty 👍

**Pull request merged**